### DISCUSSION TITLE
[Show Off] Claude 3 WIP

### DISCUSSION THREAD

**Discussion by H0llyW00dzZ** [ChatGPTNextWeb/NextChat#4269] [Show and tell]
> [!NOTE]  
> This Implementing Related to this:
> - #4220 
> - #2813 
>
> Also This Service Provider are had own for `Anthropic` and stable smiliar like `OpenAI` Made by @Yidadaa, plus already support `one-api` and `litellm`
>
> ETA: `Next Month After EID Mubarak`

https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/f8c9a52d-afb8-40c1-8e23-8081183858ba

**Comment by H0llyW00dzZ** [comment]
Also, special thanks to my friends who provided me with the key to implement this.

**Comment by H0llyW00dzZ** [comment]
### Additional Note

> [!NOTE]  
> This AI/Model does not support a role system. It only supports the assistant and user roles:
>
> ![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/fbb04200-b5e7-4acd-8e69-1b49591835b1)
>
> ![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/da4f318c-d2f8-4579-aef2-f39fc7fbba12)

**Comment by jdassd** [comment]
how to use it?

**Reply by H0llyW00dzZ** [reply]
This feature is not yet available. It's currently only available in my local development build since I haven't pushed it to GitHub.

**Reply by jdassd** [reply]
Thanks for the reply, looking forward to this feature.

**Reply by H0llyW00dzZ** [reply]
> Thanks for the reply, looking forward to this feature.

However, it needs refactoring for Claude, especially for the concatenated send messages logic in this one:

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/2560ac49-f39e-4a52-b01d-fa69c4c1ec3b)

Claude is not currently supported for that functionality.

**Reply by jdassd** [reply]
It's hard work.

**Reply by H0llyW00dzZ** [reply]
@jdassd Yes, because I'm trying hard to start from scratch and make it better instead of currently only expecting the `OpenAI` model, which doesn't support other models that are not available in `OpenAI`.

**Reply by H0llyW00dzZ** [reply]
Also it's so stable with [`one-api`](https://github.com/songquanpeng/one-api) I've been testing with my friends

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/a4260e02-9c42-45c9-b533-062ffb524d29)
![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/3d33c809-b91c-48b7-b1c5-ceb533577f5e)

**Reply by jdassd** [reply]
Will this feature arrive in the next version?

**Reply by H0llyW00dzZ** [reply]
@jdassd not sure, because of time

**Comment by H0llyW00dzZ** [comment]
bump testing `one-api` on my k8s and it stable

![k8s](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/d2b4aa17-b696-4ab2-bd57-05073f74202d)

**Comment by H0llyW00dzZ** [comment]
bump this stable just like sdk/package hahahaha

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/6687199a-45eb-4ff2-890c-add555edac6a)
